location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,577 @@
+
+    Calls the youtube-dl tool to download a YouTube video by its video_id and
+    convert the video into an audio file. If the download is successful,
+    it stores the video_id in a list, downloaded_video.
+
+    Args:
+        dest_dir: Path to the parent directory where the downloaded videos are
+            to be stored.
+        video_id: the video_id of the YouTube video to be downloaded.
+        failed_downloads: a list where failed downloaded YouTube videos'
+            video_id will be stored.
+        redo: A boolean of specifying whether to re-download all the YouTube
+            videos.
+
+    Returns:
+        A boolean of whether the download was successful or not. Returns false
+        if the video was already downloaded and redo is False.
+    """
+    check_dir(dest_dir)
+    # check to see if the video_id has already been downloaded and skip the
+    # download if it is already downloaded unless the -r, --redo flag has been
+    # passed
+    video_path = join(dest_dir, 'yt_videos', 'sliced_' + video_id + '.wav')
+    already_downloaded = isfile(video_path)
+    if already_downloaded and not redo:
+        logging.info('Already Downloaded')
+        return False
+
+    ydl_opts = {
+        'format': 'bestaudio/best',
+        'postprocessors': [{
+            'key': 'FFmpegExtractAudio',
+            'preferredquality': '192',
+        }],
+        # Force the file naming of outputs.
+        'outtmpl': join(dest_dir, 'tmp', video_id + '.%(ext)s')
+    }
+    with youtube_dl.YoutubeDL(ydl_opts) as ydl:
+        try:
+            ydl.download(['https://www.youtube.com/watch?v=' + video_id])
+            logging.info('Download Complete')
+            return True
+        except youtube_dl.DownloadError:
+            store_failed_download(dest_dir, video_id)
+            failed_downloads.append(video_id)
+            logging.info('Downloading Failed.')
+            return False
+
+
+def chop_audio(dest_dir, video_id, start_time, end_time):
+    """Chops an audio file into a segment by a given start_time and end_time.
+
+    Using a specific start_time and end_time in seconds, it chops the audio file
+    from the downloaded YouTube video, and then removes the original audio file.
+
+    Args:
+        dest_dir: Path to the parent directory where the downloaded videos are
+            are to be stored.
+        video_id: The video_id of the audio from the downloaded YouTube video.
+        start_time: The start time in seconds of the labelled video segment.
+        end_time: The end time in seconds of the labelled video segment.
+    """
+    tmp_path = join(dest_dir, 'tmp')
+    if isdir(tmp_path):
+        length = len(listdir(tmp_path))
+        m4a_path = join(tmp_path, video_id + '.m4a')
+        opus_path = join(tmp_path, video_id + '.opus')
+        ogg_path = join(tmp_path, video_id + '.ogg')
+        wav_path = join(dest_dir, 'yt_videos', 'sliced_' + video_id + '.wav')
```

Comment:
Why are there different video files?

Reply:
Depending on how the video was recorded, the audio may be downloaded in different formats. Here, I am trying to account for all the formats I've seen so far.